src/plugins/inputs/msg.js

```diff
@@ -1,9 +1,71 @@
 "use strict";
 
-exports.commands = ["msg", "say"];
+const Chan = require("../../models/chan");
+const Msg = require("../../models/msg");
+
+exports.commands = ["query", "msg", "say"];
+
+function parseInput(cmd, args, chan) {
+	switch (cmd) {
+	case "msg":
+	case "query":
+		return args.shift();
+	default:
+		return chan.name;
+	}
+}
 
 exports.input = function(network, chan, cmd, args) {
-	const target = cmd === "msg" ? args.shift() : chan.name;
+	const target = parseInput(cmd, args, chan);
+
+	if (cmd === "query") {
+		if (target.length === 0) {
+			chan.pushMessage(this, new Msg({
+				type: Msg.Type.ERROR,
+				text: "You cannot open a query window without an argument.",
+			}));
+			return;
+		}
+
+		const query = network.getChannel(target);
+		const isQueryOpened = (typeof query) !== "undefined";
+
+		if (!isQueryOpened) {
```

Comment:
```suggestion
		if (typeof query === "undefined") {
```

We never do it the other way in our codebase.

Reply:
The Idea was to make it more readable

Reply:
It ends up being less readable though. It's being negated twice.